crs_application/src/task_managers/motion_planning_manager.cpp

```diff
@@ -149,33 +177,52 @@ common::ActionResult MotionPlanningManager::splitToolpaths()
       if(current_dist + d > split_dist)
       {
         RCLCPP_INFO(node_->get_logger(),"Found split at point %lu of raster %lu",p_idx, r_idx);
-        split_point_idx = p_idx;
+        rasters_breakpoints[r_idx].push_back(p_idx);
         current_dist = 0;
-        break;
       }
       current_dist += d;
     }
+  }
 
-    if(split_point_idx >=0)
+  // splitting toolpath
+  toolpaths_processes.resize(1);
+  for(std::size_t r_idx = 0; r_idx < rasters_breakpoints.size(); r_idx++)
+  {
+    geometry_msgs::msg::PoseArray& raster = input_->rasters[r_idx];
+    auto& breakpoints_indices = rasters_breakpoints[r_idx];
+    if(breakpoints_indices.empty())
     {
-      std::remove_reference <decltype(raster)>::type new_raster;
-      std::copy(raster.poses.begin(), raster.poses.begin() + split_point_idx,
-                std::back_inserter(new_raster.poses));
-      toolpaths_processes.back().rasters.push_back(new_raster);
+      toolpaths_processes.back().rasters.push_back(raster);
+      continue;
+    }
 
-      // copying remaining part of raster into a new  process
-      toolpaths_processes.resize(toolpaths_processes.size() + 1);
+    std::size_t start_idx = 0;
+    std::size_t end_idx;
+    std::remove_reference <decltype(raster)>::type new_raster;
+    for(std::size_t p_idx = 1; p_idx < breakpoints_indices.size(); p_idx++)
+    {
+      // copy portion of raster leading up to breakpoint into current toolpath
+      end_idx = breakpoints_indices[p_idx];
       new_raster.poses.clear();
-      std::copy(raster.poses.begin() + split_point_idx + 1, raster.poses.end(),
+      std::copy(raster.poses.begin() + start_idx, raster.poses.begin() + end_idx,
                 std::back_inserter(new_raster.poses));
-      toolpaths_processes.back().rasters.push_back(new_raster);
+      toolpaths_processes.back().rasters.push_back(raster);
```

Comment:
I believe you meant to put 
```suggestion
      toolpaths_processes.back().rasters.push_back(new_raster);
```

Reply:
yes good catch